apps/mosaico/src/Analytics/Analytics.purs

```diff
@@ -0,0 +1,67 @@
+module Mosaico.Analytics where
+
+import Prelude
+
+import Effect (Effect)
+import Effect.Uncurried (EffectFn1, runEffectFn1)
+import Data.Maybe (Maybe(..), fromMaybe)
+import Lettera.Models (Author, Tag)
+import Data.DateTime (DateTime)
+import KSF.User.Cusno (toString, Cusno)
+import KSF.Api.Subscription (Subscription)
+import Data.Foldable (foldMap)
+import Data.Formatter.DateTime (format)
+import KSF.Helpers (dateTimeFormatter)
+import Data.Array (intercalate)
+
+
+type ArticleMetadata = 
+  { title :: String
+  , publishingTime :: Maybe DateTime
+  , authors :: Array Author
+  , premium :: Boolean
+  , listTitle :: Maybe String
+  , category :: Maybe String
+  , section :: Maybe String
+  , articleUuid :: String
+  , tags :: Array Tag
+  , userCusno :: Maybe Cusno
+  , userSubs :: Maybe (Array Subscription)
+  }
+
+type StringArticleMetadata = 
+  { title :: String
+  , publishingTime :: String
+  , authors :: String
+  , premium :: String
+  , listTitle :: String
+  , category :: String
+  , section :: String
+  , articleUuid :: String
+  , tags :: String
+  , userCusno :: String
+  , userSubs :: String
+  }
+
+foreign import _pushToDataLayer :: EffectFn1 StringArticleMetadata Unit
+pushToDataLayer :: ArticleMetadata -> Effect Unit
+pushToDataLayer = runEffectFn1 _pushToDataLayer <<< stringifyMetadata
+
+stringifyMetadata :: ArticleMetadata -> StringArticleMetadata
+stringifyMetadata metadata = 
+  { title : metadata.title
+  , publishingTime: foldMap (\x -> format dateTimeFormatter x) metadata.publishingTime
+  , premium: show metadata.premium
+  , category: fromMaybe "" metadata.category
```

Comment:
In these kind of situations we could use `fold` (https://pursuit.purescript.org/packages/purescript-foldable-traversable/5.0.1/docs/Data.Foldable#v:fold), which, when given a `Maybe String`, returns the underlying `String`, or if we don't have a value, then `mempty` is returned (which evaluates to `""`).

Reply:
This is fine either way though!